src/python/gudhi/tensorflow/ripsnet.py

```diff
@@ -0,0 +1,209 @@
+        for layer in self.layers:
+            outputs = layer(outputs)
+        return outputs
+
+
+class PermopRagged(tf.keras.layers.Layer):
+    """
+    This is a class for the permutation invariant layer in the RipsNet architecture.
+    """
+
+    def __init__(self, perm_op, **kwargs):
+        """
+        Constructor for the PermopRagged class.
+
+        Parameters:
+            perm_op: permutation invariant function, such as `tf.math.reduce_sum`, `tf.math.reduce_mean`.
+        """
+        super().__init__(dynamic=True, **kwargs)
+        self._supports_ragged_inputs = True
+        self.pop = perm_op
+
+    def build(self, input_shape):
+        super().build(input_shape)
+
+    def call(self, inputs):
+        """
+        Apply PermopRagged on an input tensor.
+        """
+        out = self.pop(inputs, axis=1)
+        return out
+
+
+class RipsNet(tf.keras.Model):
+    """
+    This is a TensorFlow model for estimating vectorizations of persistence diagrams of point clouds.
+    This class implements the RipsNet described in the following article <https://arxiv.org/abs/2202.01725>.
+    """
+
+    def __init__(self, phi_1, phi_2, input_dim, perm_op='mean', **kwargs):
+        """
+        Constructor for the RipsNet class.
+
+        Parameters:
+            phi_1 (layers): any block of DenseRagged layers. Can be :class:`~gudhi.tensorflow.DenseRaggedBlock`, or a custom block built from :class:`~gudhi.tensorflow.DenseRagged` layers.
+            phi_2 (layers): Can be any (block of) TensorFlow layer(s),  e.g. :class:`~gudhi.tensorflow.TFBlock`.
+            input_dim (int): dimension of the input point clouds.
+            perm_op (str): Permutation invariant operation. Can be 'mean' or 'sum'.
+        """
+        super().__init__(dynamic=True, **kwargs)
+        self.phi_1 = phi_1
+        self.pop = perm_op
+        self.phi_2 = phi_2
+        self.input_dim = input_dim
+
+        if perm_op not in ['mean', 'sum']:
+            raise ValueError(f'Permutation invariant operation: {self.pop} is not allowed, must be "mean" or "sum".')
+
+    def build(self, input_shape):
+        return self
+
+    def call(self, pointclouds):
+        """
+        Apply RipsNet on a ragged tensor containing a list of pointclouds.
+
+        Parameters:
+            point clouds (n x None x input_dimension): ragged tensor containing n pointclouds in dimension `input_dimension`. The second dimension is ragged since point clouds can have different numbers of points.
+
+        Returns:
+            output (n x output_shape): tensor containing predicted vectorizations of the persistence diagrams of pointclouds.
+        """
+        if self.pop == 'mean':
+            pop_ragged = PermopRagged(tf.math.reduce_mean)
+        elif self.pop == 'sum':
+            pop_ragged = PermopRagged(tf.math.reduce_sum)
+        else:
+            raise ValueError(f'Permutation invariant operation: {self.pop} is not allowed, must be "mean" or "sum".')
```

Comment:
Well, the only concern I have is that if we leave this entirely up to the user, then their input function may be such that our theoretical guarantees for RipsNet may perhaps no longer be satisfied. I'm not sure what the best option is in this case. What do you think?

Reply:
I think it's good to let the user do whatever they want (as long as it is properly documented). Our theoretical results are of the form "if ..., then" ; but nothing prevents to use RipsNet with some other `perm_op` and a user may be interested in doing so.

Reply:
Okay, sounds good, I've removed the requirement that the permutation invariant function has to be 'mean' or 'sum', so that users can specify their own functions.